aws-synthetics-canary/src/test/java/com/amazon/synthetics/canary/DeleteHandlerTest.java

```diff
@@ -1,58 +1,172 @@
 package com.amazon.synthetics.canary;
 
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.Mock;
-import org.mockito.junit.jupiter.MockitoExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import software.amazon.awssdk.services.synthetics.model.Canary;
+import software.amazon.awssdk.services.synthetics.model.CanaryState;
+import software.amazon.awssdk.services.synthetics.model.CanaryStatus;
+import software.amazon.awssdk.services.synthetics.model.ConflictException;
+import software.amazon.awssdk.services.synthetics.model.DeleteCanaryRequest;
+import software.amazon.awssdk.services.synthetics.model.GetCanaryRequest;
+import software.amazon.awssdk.services.synthetics.model.GetCanaryResponse;
+import software.amazon.awssdk.services.synthetics.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.synthetics.model.StopCanaryRequest;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnResourceConflictException;
 import software.amazon.cloudformation.proxy.*;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
-@ExtendWith(MockitoExtension.class)
-public class DeleteHandlerTest extends TestBase{
-    DeleteHandler handler;
+public class DeleteHandlerTest extends TestBase {
+    private static final String CANARY_NAME = "canary-name";
+    private static final ResourceHandlerRequest<ResourceModel> REQUEST = ResourceHandlerRequest.<ResourceModel>builder()
```

Comment:
This works for now.. but generally using setup is preferred so as to ensure that the tests do not influence each other. For eg, if there was another param of this request, that got modified during a call, the next test that uses it.. will have a corrupted request (unintentionally)

Reply:
JUnit 5 uses a new test class instance for each test method, so this is not a concern in this case.